core/settings/qPrefDiveComputer.h

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0
+#ifndef QPREFSDIVECOMPUTER_H
+#define QPREFSDIVECOMPUTER_H
+#include "core/pref.h"
+
+#include <QObject>
+
+class qPrefDiveComputer : public QObject {
+	Q_OBJECT
+	Q_PROPERTY(QString device READ device WRITE set_device NOTIFY device_changed);
+	Q_PROPERTY(QString device_name READ device_name WRITE set_device_name NOTIFY device_name_changed);
+	Q_PROPERTY(int download_mode READ download_mode WRITE set_download_mode NOTIFY download_mode_changed);
+	Q_PROPERTY(QString product READ product WRITE set_product NOTIFY product_changed);
+	Q_PROPERTY(QString vendor READ vendor WRITE	set_vendor NOTIFY vendor_changed);
+
+public:
+	qPrefDiveComputer(QObject *parent = NULL);
+	static qPrefDiveComputer *instance();
+
+	// Load/Sync local settings (disk) and struct preference
+	void loadSync(bool doSync);
+	void load() {loadSync(false); }
+	void sync() {loadSync(true); }
+
+public:
+	static inline const QString device() {return QString(prefs.dive_computer.device); };
+	static inline const QString device_name() {return QString(prefs.dive_computer.device_name); };
+	static inline int download_mode() {return prefs.dive_computer.download_mode; };
+	static inline const QString product() {return QString(prefs.dive_computer.product); };
+	static inline const QString vendor() {return QString(prefs.dive_computer.vendor); };
+
+public slots:
+	void set_device(const QString& device);
+	void set_device_name(const QString& device_name);
+	void set_download_mode(int mode);
+	void set_product(const QString& product);
+	void set_vendor(const QString& vendor);
+
+signals:
+	void device_changed(const QString& device);
```

Comment:
We put '&' and '*' to the variable name:
```
	void device_changed(const QString &device);
```
Some people don't like it - but that's how the C-inherited grammar works - sorry.

Reply:
This is not a matter of taste, you talk about 2 different situations
The first one &var is old C style, used in statement like x=&y meaning “take address of”
The second is C++ function declaration style, meaning “get reference to” (so you avoid calling with func(&var)

I am aware the actual assembler code generated is the same, but the meaning is different, which is why it is normal to use
type& var in function declarations. 
as an anekdote, this was an idiologic fight to Bjarne, who wanted another symbol to mark the difference clearly.

Reply:
well, technically it IS a matter of taste, but I'm ok with adding this as a different category to the coding style document.

Reply:
Please no - with a few exceptions [which I'd like to fix] we use the `*` and `&`  stick to the variable name convention. Putting it to the type is just against reality. From a parser point of view, type declarations with `*` and `&` are the same. That the original elegance of C [`int *i` meaning the object pointed to by `i` is an `int` is lost for reference notation - `int &i` does not mean the address of `i` is an `int`] is irrelevant. Either we choose one or the other, but please not a mix based on dubious reasoning. Consider the line:
```
int&     i=x, j=y;
```
`i` is a reference to int and `j` is an int, which is in contradiction to what the line implies. Sorry, but that's the way it works. I have a feeling that people using the `int& i` syntax are in denial and can't accept the fact that C++, by inheritance from C, works the way it works. To those people, I would recommend moving to a "pure" language of their liking, such as rust.

So for me it's not a big deal if some isolated parts of the code violate our conventions, but let's not make it "official" and end up with a random mess. The code is inconsistent enough as it is.

Reply:
you make examples in east while I make code in west! I wrote and mean how to use & (get by reference) in function declarations.

There is a difference between “call by address” and “reference by address”, fell free to take a look at the parser. The most obvious difference is that the caller do not use & 2 examples:
foo(&x).     void foo(int *x).    this is the traditional C way.
foo(x)           void foo(int& x). is the more elegant (some see it is confusing) way, which was introduced with C++. and you will see good C++ compilers like clang use the difference, so that is the reason to also write it.

I DO NOT talk about statements, where it makes no sense to put the & on the type.

I do not care what your job is, you are surely the local C++ expert in this project, but that hardly makes you the only expert, who have the right to give c++ lessons (as you wrote you fell its needed). I personally have spent many years working on C++ backends, preprocessors and parsers (actually much less than writing application code). I do NOT comment other peoples code on “relidious” issues, nor do I give lessons in C++

I feel for @dirkhh with his statement about reading PR´s and I am sorry for ince again being the cause, but I think everyone gets a bit edgy when ones professional experience is reduced by having to take C++ lessons. You are both collaborators so nothing hinders you from changing the code once (if) this PR is merged (you have the capacity to do it in this PR, but I would see that as a not nice way to make a point).

@dirkhh have pointed at a couple of RELEVANT changes, which will be uploaded in a couple of hours, apart from that I will stand back and watch.


I am sorry writing personally in this comment, I have tried hard to not to be personal (I do NOT wish to offend anyone)j, but my professionalidm is in question here, something I cannot just neglect.

Reply:
Please, stop it.

You fail to acknowledge that `&`, `*` (and recently `&&`) have *different* meaning in expression and type declaration context. You keep mixing it up. Even *if* you prefer to put the `&`, `*` and `&&` *declarators* to the type, then you write
```
void qPrefPrivate::copy_txt(const char** name, const QString& string)
```
If you don't, then you write
```
void qPrefPrivate::copy_txt(const char **name, const QString &string)
```
A mixture of both makes no sense. Not from a parser, not from a language, not from any perspective.

I'm not judging anyone for language-savviness. It doesn't matter to me at all as long as the contributions are good. In my opinion, the only thing that may "reduce your professional experience" is your over-reaction to criticism.

Reply:
I think we are mixing a coding style (=subjective, personal) discussion with a language semantics discussion.

I looked at this again and I think I would like us to be consistent across both declaration and use, across both reference, address, and dereference. The designator should tie to the variable or argument name, not to the type.

It's easy enough to handle this in cleanup - but from a project coding style perspective, I would prefer this consistency.

This isn't about anyone being right or wrong, this is the same logic by which I request that we use tabs and not spaces and that we write if/else clauses a certain way.

Reply:
This is *just* a matter of style, the code will work the same, there's no difference *at all*, the assembly generated is completely the same. with all that said, stop. Some people will argue that the correct way is to use QString const &, and again, the assembly generated is the same.